src/instruments/view/instrumentlistmodel.cpp

```diff
@@ -259,9 +289,16 @@ void InstrumentListModel::selectInstrument(const QString& instrumentId, const QS
     emit selectedInstrumentsChanged();
 }
 
-void InstrumentListModel::makeSoloist(const QString&)
+void InstrumentListModel::toggleSoloist(const QString& instrumentId)
 {
-    NOT_IMPLEMENTED;
+    for (int i = 0; i < m_selectedInstruments.count(); ++i) {
+        if (m_selectedInstruments[i].id == instrumentId) {
```

Comment:
I think, we need to add a new method for getting an index of a selected instrument by id:

int selectedInstrumentIndex(instrumentId)

Because the code for getting the selected instrument can be found in two different places (toggleSoloist and unselectInstrument)

Reply:
Made a private method `InstrumentListModel::findInstrumentIndex(const QString& instrumentId) const` which returns the index in `m_selectedInstruments` of the specified `instrumentId` and this method is used in both `toggleSoloist()` and `unselectInstrument()`.